database/pgx/pgx.go

```diff
@@ -85,6 +86,24 @@ func WithInstance(instance *sql.DB, config *Config) (database.Driver, error) {
 		config.DatabaseName = databaseName
 	}
 
+	if len(config.MigrationsTable) == 0 {
+		config.MigrationsTable = DefaultMigrationsTable
+	}
+
+	config.SchemaName = ""
+	if config.MigrationsTableQuoted {
+		re := regexp.MustCompile(`"(.*?)"`)
+		result := re.FindAllStringSubmatch(config.MigrationsTable, -1)
+		config.TableName = result[len(result)-1][1]
+		if len(result) == 2 {
+			config.SchemaName = result[0][1]
```

Comment:
We shouldn't set `config.SchemaName` using the schema found in `config.MigrationsTable` since 2 different schemas could be set by both `config.SchemaName` and `config.MigrationsTable`. e.g. `config.SchemaName = "my_schema1"` and `config.MigrationsTable = "\"my_schema2\".\"migrations_table\"`

We should probably add a `migrationsSchemaName` field to the `Config` struct and set that here. Its default value should be the current schema.